graphql-dgs-client/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/GraphQLQueryRequest.kt

```diff
@@ -63,10 +57,40 @@ class GraphQLQueryRequest(private val query: GraphQLQuery, private val projectio
         if (projection is BaseSubProjectionNode<*, *>) {
             builder.append(projection.root().toString())
         } else if (projection != null) {
+
             builder.append(projection.toString())
         }
 
         builder.append(" }")
         return builder.toString()
     }
 }
+
+class InputValueSerializer(private val scalars: Map<Class<*>, Coercing<*, *>> = emptyMap()) {
+    fun serialize(input: Any?): String? {
+        if (input == null) {
+            return null
+        }
+
+        val type = input::class.java
+        return if (scalars.contains(type)) {
+            """"${scalars[type]!!.serialize(input)}""""
+        } else if (type.isPrimitive || type == Integer::class.java || type == Long::class.java || type == Double::class.java || type == Float::class.java || type == Boolean::class.java || type == Short::class.java || type == Byte::class.java || type.isEnum) {
+            input.toString()
+        } else if (type == String::class.java) {
+            """"$input""""
+        } else if (input is List<*>) {
+            """[${input.filterNotNull().joinToString(", ") { listItem -> serialize(listItem) ?: "" }}]"""
+        } else {
+            input.javaClass.declaredFields.map {
+                it.isAccessible = true
+                val nestedValue = it.get(input)
```

Comment:
Does this null check work if the field is an unboxed type, e.g. int?

Reply:
primitives can't be null, they always have a default value instead.

Reply:
What I mean is do you need to do the null check only if the type is not primitive? because it is an incompatible check, no?

Reply:
I moved the null check after updating the implementation, so this isn't relevant anymore